Sources/Sentry/SentrySamplingProfiler.cpp

```diff
@@ -107,8 +107,8 @@ namespace profiling {
         }
         isSampling_ = true;
         numSamples_ = 0;
-        thread_ = std::thread(samplingThreadMain, port_, clock_, delaySpec_, cache_, callback_,
-            std::ref(numSamples_), onThreadStart);
+        thread_ = std::thread(samplingThreadMain, port_, clock_, delaySpec_, std::cref(cache_),
```

Comment:
TIL `std::ref` and `std::cref`, nice! I was trying to do this with `&` which wouldn't compile 😭